modules/light-clients/11-beefy/types/header.go

```diff
@@ -0,0 +1,103 @@
+package types
+
+import (
+	"bytes"
+	"encoding/gob"
+	"time"
+
+	substrateTypes "github.com/centrifuge/go-substrate-rpc-client/types"
+	clienttypes "github.com/cosmos/ibc-go/v3/modules/core/02-client/types"
+	"github.com/cosmos/ibc-go/v3/modules/core/exported"
+)
+
+var _ exported.Header = &Header{}
+
+const revisionNumber = 0
+
+// DecodeParachainHeader decodes an encoded substrate header to a concrete Header type. It takes encoded bytes
+// as an argument and returns a concrete substrate Header type.
+func DecodeParachainHeader(hb []byte) (substrateTypes.Header, error) {
+	h := substrateTypes.Header{}
+	dec := gob.NewDecoder(bytes.NewReader(hb))
+	err := dec.Decode(&h)
+	if err != nil {
+		return substrateTypes.Header{}, err
+	}
+	return h, nil
+}
+
+// DecodeExtrinsicTimestamp decodes a scale encoded timestamp to a time.Time type
+func DecodeExtrinsicTimestamp(encodedExtrinsic []byte) (time.Time, error) {
+	extrinsic := substrateTypes.Extrinsic{}
+	dec := gob.NewDecoder(bytes.NewReader(encodedExtrinsic))
+	err := dec.Decode(&extrinsic)
+	if err != nil {
+		return time.Time{}, err
+	}
+
+	var timestamp int64
+	timestampDecode := gob.NewDecoder(bytes.NewReader(extrinsic.Method.Args))
+	err = timestampDecode.Decode(&timestamp)
+	if err != nil {
+		return time.Time{}, err
+	}
+
+	return time.Unix(timestamp, 0), nil
+}
+
+// ConsensusState returns the updated consensus state associated with the header
+func (h Header) ConsensusState() *ConsensusState {
+	parachainHeader, err := DecodeParachainHeader(h.ParachainHeader)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	rootHash, err := parachainHeader.StateRoot.MarshalJSON()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	timestamp, err := DecodeExtrinsicTimestamp(h.TimestampExtrinsic)
```

Comment:
we  need to verify that this extrinsic in `extrinisic_root` using https://github.com/ChainSafe/gossamer/blob/e1f7f96cb8cb7e82156f7333b9a0704ab8151fd2/lib/trie/proof.go?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L74